localstack/services/kms/models.py

```diff
@@ -247,27 +244,34 @@ def verify(
         signing_algorithm: SigningAlgorithmSpec,
         signature: bytes,
     ) -> bool:
-        kwargs = self._construct_sign_verify_kwargs(signing_algorithm)
-        if message_type != MessageType.DIGEST:
-            data = self._get_digest(data)
+        kwargs = self._construct_sign_verify_kwargs(signing_algorithm, message_type)
         try:
             self.crypto_key.key.public_key().verify(signature=signature, data=data, **kwargs)
             return True
+        except ValueError as exc:
+            raise ValidationException(str(exc))
         except InvalidSignature:
             # AWS itself raises this exception without any additional message.
             raise KMSInvalidSignatureException()
 
-    def _construct_sign_verify_kwargs(self, signing_algorithm: SigningAlgorithmSpec) -> Dict:
+    def _construct_sign_verify_kwargs(
+        self, signing_algorithm: SigningAlgorithmSpec, message_type: MessageType
+    ) -> Dict:
         kwargs = {}
 
         if "SHA_256" in signing_algorithm:
-            kwargs["algorithm"] = hashes.SHA256()
+            hasher = hashes.SHA256()
         elif "SHA_384" in signing_algorithm:
-            kwargs["algorithm"] = hashes.SHA384()
+            hasher = hashes.SHA384()
         elif "SHA_512" in signing_algorithm:
-            kwargs["algorithm"] = hashes.SHA512()
+            hasher = hashes.SHA512()
+        else:
+            LOG.exception("Unsupported hash type in SigningAlgorithm '%s'", signing_algorithm)
```

Comment:
This case would raise `NameError` further below, as `hasher` is not defined. Should we rather raise this as an exception here?

Reply:
Fixed :+1: